### ISSUE TITLE
how to train 

### ISSUE THREAD

**Issue by Ericchen328** [lpiccinelli-eth/UniDepth#4]
I have my own dataset and want to train this model on my own dataset but i didn‘t see the detail of how to train

**Comment by jhoon-kim** [comment]
First of all, thank you for writing a good paper and releasing the code.
But I'm having trouble implementing the training code based on just what you said.

1. I understood that SILog is the loss corresponding to z_log, `MSE of camera prediction loss is the loss corresponding to ray and angle in the code, and SelfCons is the loss of two depth features with different augmentations applied. Is that correct?
2. There is a mention of bi-directional loss in "3.3. Geometric Invariance Loss". Is it included in the final optimization of 3.4?
3. The implementation of the function in (4) does not appear to exist in `losses.py`. Does MSE play the same role? The only class that uses variance is SILog.
4. The init of SILog and SelfCons requires a `weight`, and the forward of SelfCons requires a `metas` argument. I did not clearly understand what the values ​​are. I couldn't find the part that assigns "K_target" or "flip", which I expected to be included in `metas`.
5. I think 3d output must be in spherical space to calculate the loss, but the "points" returned by the forward of the unidepth class appear to be euclidean. Should I convert this back to spherical to calculate the loss?
6. Does the geometric and appearance augmentation in the implementation details of “4. Experiments” apply to both views?
7. The gradient of k and v entering the camera module should not be transmitted, but I am not sure exactly which part of the code it is.
8. I don't know how to implement the pipeline correctly to obtain o1, o2, d2|e2, and d1|e1.

Sorry for the many questions, but I want to try using it for my custom data.
Can you please tell me the expected schedule for the release of the training code or provide answers to the above issues?
Thank you in advance.

**Comment by CDchenlin** [comment]
Thank you for your great work. I wonder if you have a timeline for the training code release?

**Comment by lpiccinelli-eth** [comment]
The code for training has been made public in the PR #76 , although still not 100% sure it will run smoothly, you can try it out.
